rotkehlchen/exchanges/bitpanda.py

```diff
@@ -0,0 +1,537 @@
+                request_url=request_url,
+                options=options,
+            )
+            try:
+                response = self.session.get(request_url, timeout=DEFAULT_TIMEOUT_TUPLE)
+            except requests.exceptions.RequestException as e:
+                raise RemoteError(f'Bitpanda API request failed due to {str(e)}') from e
+
+            if response.status_code == HTTPStatus.TOO_MANY_REQUESTS:
+                backoff_in_seconds = int(20 / retries_left)
+                retries_left -= 1
+                log.debug(
+                    f'Got a 429 from Bitpanda query of {request_url}. Will backoff '
+                    f'for {backoff_in_seconds} seconds. {retries_left} retries left',
+                )
+                gevent.sleep(backoff_in_seconds)
+                continue
+
+            if response.status_code != HTTPStatus.OK:
+                raise RemoteError(
+                    f'Bitpanda API request failed with response: {response.text} '
+                    f'and status code: {response.status_code}',
+                )
+
+            # we got it, so break
+            break
+
+        else:  # retries left are zero
+            raise RemoteError(f'Ran out of retries for Bitpanda query of {request_url}')
+
+        try:
+            decoded_json = jsonloads_dict(response.text)
+        except json.decoder.JSONDecodeError as e:
+            raise RemoteError(f'Invalid JSON {response.text} in Bitpanda response. {e}') from e
+
+        if 'data'not in decoded_json:
+            raise RemoteError(
+                f'Invalid JSON {response.text} in Bitpanda response. Expected "data" key',
+            )
+
+        return decoded_json['data'], decoded_json.get('meta'), decoded_json.get('links')
+
+    @overload
+    def _query_endpoint_until_end(  # pylint: disable=no-self-use
+            self,
+            endpoint: Literal['trades'],
+            from_ts: Optional[Timestamp],
+            to_ts: Optional[Timestamp],
+            options: Optional[Dict[str, Any]] = None,
+    ) -> List[Trade]:
+        ...
+
+    @overload
+    def _query_endpoint_until_end(  # pylint: disable=no-self-use
+            self,
+            endpoint: Literal['fiatwallets/transactions'],
+            from_ts: Optional[Timestamp],
+            to_ts: Optional[Timestamp],
+            options: Optional[Dict[str, Any]] = None,
+    ) -> List[AssetMovement]:
+        ...
+
+    def _query_endpoint_until_end(
+            self,
+            endpoint: Literal['trades', 'fiatwallets/transactions'],
+            from_ts: Optional[Timestamp],
+            to_ts: Optional[Timestamp],
+            options: Optional[Dict[str, Any]] = None,
+    ) -> Union[List[Trade], List[AssetMovement]]:
+        """Query a paginated endpoint until all pages are read
+
+        May raise RemoteError
+        """
+        from_timestamp = from_ts if from_ts is not None else Timestamp(0)
+        to_timestamp = to_ts if to_ts is not None else ts_now()
```

Comment:
I don't know if this needs to be checked here but should we check that `to_timestamp > from_timestamp`?

Reply:
We don't do that anywhere where we use from_ts and to_ts. If at any point we do it, it should be in the marshmallow checker imo